### PR TITLE
chore(ci): cleanup gha workflows

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

This is a general cleanup/sanitation of gha workflows. It bumps some 3rd party workflows to latest and ensures we only reference them using commit shas (per [github's recommendations](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions)). Additionally, it moves our usage of the devtoolsbot user to the github app.

### CODE REVIEW

**Review by himanshusinghs — APPROVED** [review]

File: .editorconfig [hunk 1]
```diff
@@ -203,4 +203,6 @@ ij_yaml_keep_indents_on_empty_lines = false
 ij_yaml_keep_line_breaks = true
 ij_yaml_space_before_colon = false
 ij_yaml_spaces_within_braces = true
-ij_yaml_spaces_within_brackets = true
\ No newline at end of file
+ij_yaml_spaces_within_brackets = true
+tab_width = 2
```
Comment (kmruiz) [inline .editorconfig:207]: Is this change on purpose? It will change the tab size of all our code, not only YAML.

**Review by kmruiz — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 1 above)
Comment (nirinchev) [reply .editorconfig:207]: Hm... that's in the yaml/yml section though - does intellij not respect the extensions in .editorconfig?

Comment (kmruiz) [reply .editorconfig:207]: Ah wait, you are totally right. Didn't see that part of the diff, was collapsed. All good!

**Review by kmruiz — APPROVED** [review]
